build.assets/tooling/cmd/notarize-apple-binaries/gonConfig.go

```diff
@@ -0,0 +1,170 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"debug/macho"
+	"encoding/binary"
+	"flag"
+	"io"
+	"os"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type GonConfig struct {
+	AppleUsername string
+	ApplePassword string
+	BinaryPaths   []string
+}
+
+func NewGonConfig() *GonConfig {
+	gc := &GonConfig{}
+	flag.StringVar(&gc.AppleUsername, "apple-username", "", "Apple Connect username used for notarization")
+	flag.StringVar(&gc.ApplePassword, "apple-password", "", "Apple Connect password used for notarization")
+
+	return gc
+}
+
+func (gc *GonConfig) Check() error {
+	err := gc.validateAppleUsername()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-username flag")
+	}
+
+	err = gc.validateApplePassword()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-password flag")
+	}
+
+	err = gc.validateBinaryPaths()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate binary path(s)")
+	}
+
+	return nil
+
+	// It might be worth adding an actual login check here in the future
+}
+
+func (gc *GonConfig) validateAppleUsername() error {
+	if gc.AppleUsername == "" {
+		return trace.BadParameter("the apple-username flag should not be empty")
+	}
+
+	return nil
+}
+
+func (gc *GonConfig) validateApplePassword() error {
+	if gc.ApplePassword == "" {
+		return trace.BadParameter("the apple-password flag should not be empty")
+	}
+
+	return nil
+}
+
+func (gc *GonConfig) validateBinaryPaths() error {
+	// Check for minimum arg count and load the parameters into the struct
+	if flag.NArg() == 0 {
+		return trace.BadParameter("the path to at least one binary is required")
+	}
+	gc.BinaryPaths = flag.Args()
+
+	// Validate each path
+	binaryPaths := flag.Args()
+	for _, binaryPath := range binaryPaths {
+		logrus.Debugf("Validating binary path %q...", binaryPath)
+		err := gc.verifyFileIsValidForSigning(binaryPath)
+		if err != nil {
+			return trace.Wrap(err, "file %q failed binary validation", binaryPath)
+		}
+	}
+
+	return nil
+}
+
+// Returns an error if the file is not valid for signing
+func (gc *GonConfig) verifyFileIsValidForSigning(filePath string) error {
+	err := gc.verifyPathIsNotDirectory(filePath)
+	if err != nil {
+		return trace.Wrap(err, "filesystem path %q is a directory")
+	}
+
+	err = gc.verifyFileIsAppleBinary(filePath)
+	if err != nil {
+		return trace.Wrap(err, "file %q is not an Apple binary", filePath)
+	}
+
+	return nil
+}
+
+// Returns an error if the provided binary path is actually a directory
+func (gc *GonConfig) verifyPathIsNotDirectory(filePath string) error {
```

Comment:
Similarly, kingping has an `ExistingFile` flag type that would handle this for you.

Reply:
Updated to use kingpin